storage/Utils/FileUtils.h

```diff
@@ -54,14 +61,38 @@ namespace storage
 
 	string get_fullname() const { return path + "/" + name; }
 
-    protected:
+    private:
 
-	string path;
+	const string path;
 	string name;
 
+	/**
+	 * Either value of TMPDIR or as a fallback /tmp.
+	 */
+	static string default_path();
+
     };
 
-}
 
+    class TmpMount : public TmpDir
+    {
+
+    public:
+
+	enum class Mode { READ_ONLY, READ_WRITE };
+
+	/**
+	 * Mounts device at TmpDir(path, name_template).
```

Comment:
Throw on failure is standard in the code. A throw in a destructor will likely terminate the program and is basically never used.